src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -89,4 +89,43 @@ public static Entity createGameEntity(String gameName, DatastoreService datastor
  
     return entity;
   }
+
+  /**
+  * add user to user list in game entity, and creates a score entity for the given user and game
+  */
+  public static boolean addUserToGame(String userId, String userName, Entity gameEntity, DatastoreService datastore) {
+ 
+    if(userId == "" || userName == ""){
+      return false;
+    }
+
+    if(gameEntity == null || datastore == null){
+      log.severe("null entity or datastore"); 
+      return false;
+    }
+ 
+    ArrayList<String> userIds = (ArrayList<String>) gameEntity.getProperty("userIds");
+    if(userIds == null){
```

Comment:
What if the size of userIds is 0 because no users have been added yet. Do you still want to return false then?

Reply:
No, if the size of user ids is 0 in some cases you still want to add the user (so like if the game was just created) this is what the last test case is testing (because nothing is added to userIds after it's initialized, it's size is 0)

    ArrayList<String> userIds = new ArrayList<>();
    gameEntity.setProperty("userIds", userIds);

    boolean actual = GameUtils.addUserToGame("user1", "1", gameEntity, datastore);

    Assert.assertEquals(true, actual);